ansible_inventory_server/jujurest.py

```diff
@@ -61,14 +61,32 @@
     return status
 
 
+def juju_filter_ip_addresses(data, kwargs):
+    """Filters IP addresses using network interface name and/or subnet"""
+
+    ip_addresses = []
+
+    try:
+        interface_filter = kwargs.get('interface')
+        for interface, interface_data in data['network-interfaces']:
+            if interface_filter and interface != interface_filter:
+                continue
+
+            ip_addresses.extend(interface_data.get('ip-addresses') or [])
+
+    except (KeyError, ValueError):
+        pass
+
+    return filter_ip_addresses(ip_addresses, kwargs)
+
+
 def juju_filter_machine_info(machine, data, kwargs):
     """Keeps only useful machine information"""
     return {
         'id': machine,
         'name': data.get('display-name') or data.get('instance-id'),
         'instance_id': data.get('instance-id'),
-        'ip_addresses': filter_ip_addresses(data.get('ip-addresses', []),
-                                            kwargs),
+        'ip_addresses': juju_filter_ip_addresses(data, kwargs),
```

Comment:
data['network-interfaces'] would be enough because this is the only part of data you are using